source/Dictionary.h

```diff
@@ -19,23 +19,60 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 
 
+typedef std::vector<double>::size_type dict;
+
 // This class stores a mapping from character string keys to values, in a way
 // that prioritizes fast lookup time at the expense of longer construction time
 // compared to an STL map. That makes it suitable for ship attributes, which are
-// changed much less frequently than they are queried.
-class Dictionary : private std::vector<std::pair<const char *, double>> {
+// changed much less frequently than they are queried. For critical paths, it
+// offers direct access to the attribute values stored in a vector.
+class Dictionary : private std::vector<dict> {
```

Comment:
I think it would be _much_ clearer if instead of inheriting, we just make this second vector a field and call it something like `usedIndices` or similar. As it is now, there is some unnamed ambient vector floating around and it is not clear at all what its purpose is unless one really understands the implementation of this class. Composition should be chosen over inheritance whenever possible.